src/Cardano/Environment.hs

```diff
@@ -0,0 +1,168 @@
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE TypeApplications #-}
+
+-- |
+-- Copyright: © 2018-2019 IOHK
+-- License: MIT
+--
+-- This module contains static configuration parameters. Rather than providing
+-- and carrying around a configuration file through the application, we resolve
+-- configuration data at runtime using the available environment.
+--
+-- This gives us a flexible and portable approach to software configuration, and
+-- remove some pain from the development perspective. Prior to starting, the
+-- wallet is expected to have a few configuration parameter available. One may
+-- rely on a `.env` file to bundle configuration settings together for a given
+-- target environment.
+
+module Cardano.Environment
+    (
+    -- * Networking
+      Network(..)
+    , network
+    , ProtocolMagic(..)
+    , protocolMagic
+    ) where
+
+import Prelude
+
+import Control.Exception
+    ( Exception (..), throwIO )
+import Data.Int
+    ( Int32 )
+import Data.Text
+    ( Text )
+import Data.Text.Class
+    ( FromText (..), TextDecodingError (..), ToText (..) )
+import Fmt
+    ( Buildable (..), nameF, padLeftF, pretty )
+import GHC.Generics
+    ( Generic )
+import System.Environment
+    ( getProgName, lookupEnv )
+import System.IO.Unsafe
+    ( unsafePerformIO )
+
+import qualified Data.Text as T
+
+
+-- | Fatal exception thrown when a required ENV var is missing upon start-up.
+data ErrMissingOrInvalidEnvVar = ErrMissingOrInvalidEnvVar
+    { name :: String
+    , command :: String
+    , additionalContext :: Maybe (String, TextDecodingError)
+    }
+
+instance Show ErrMissingOrInvalidEnvVar where
+    show = displayException
+
+-- | Produces a nice terminal output so that the error is very readable.
+--
+-- @
+-- $ NETWORK=patate cardano-wallet-launcher
+-- Starting...
+-- cardano-wallet-launcher: Missing or invalid ENV var:
+--
+--     ENV[NETWORK] = patate
+--                     |
+--                     |
+--                     *--> patate is neither "mainnet", "testnet", "staging" nor "local".
+--
+-- @
+--
+-- @
+-- $ cardano-wallet-launcher
+-- Starting...
+-- cardano-wallet-launcher: Missing or invalid ENV var:
+--
+--     ENV[NETWORK] = ?
+--
+-- What about trying to provide a valid ENV var `NETWORK=value cardano-wallet-launcher` ?
```

Comment:
~will this error be presentet using `stack exec -- cardano-wallet server ....` as well? If it will be, maybe this error message will be missleading as it constrains only to `cardano-wallet-launcher` binary~

EDIT: I see command is not hardcoded - so everything is ok